File: Reliefy/Reliefy.UI/mobile/lib/models/appointment.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:mobile/models/user.dart';
part 'appointment.g.dart';

@JsonSerializable()
class Appointment {
  final String? id;
  final DateTime? startTime;
  final DateTime? endTime;
  final String? patientId;
  final User? patient;
  final String? doctorId;
  final User? doctor;
  final int? status;
  final double? score;

  Appointment(
      {this.id, this.startTime, this.endTime, this.patientId, this.patient, this.doctorId, this.doctor, this.status, this.score});
  factory Appointment.fromJson(Map<String, dynamic> json) => _$AppointmentFromJson(json);

  Map<String, dynamic> toJson() => _$AppointmentToJson(this);
}

File: Reliefy/Reliefy.Domain/Entities/Appointment.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Reliefy.Domain.Enum;

namespace Reliefy.Domain.Entities;

public class Appointment : Auditable
{
	[Key]
	[DatabaseGenerated(DatabaseGeneratedOption.Identity)]
	public Guid Id { get; set; }
	
	[Column(TypeName = "timestamp with time zone")]
	public DateTime StartTime { get; set; }
	
	[Column(TypeName = "timestamp with time zone")]
	public DateTime EndTime { get; set; }

	public AppointmentStatus Status { get; set; }
	
	public int? Score { get; set; }
	
	[ForeignKey("Patient")]
	public Guid PatientId { get; set; }

	public User Patient { get; set; }
	
	[ForeignKey("Doctor")] 
	public Guid DoctorId { get; set; }
	
	public User Doctor { get; set; }
}